### PR TITLE
🏗 Generate test reports on Travis builds.

### PR DESCRIPTION
**Pull request by Rafer45**
Closes ampproject/amp-github-apps#885.

EDIT: The Travis build passed! See (https://travis-ci.org/github/ampproject/amphtml/builds/700200481).
This PR introduces code that only runs when the branch is pulled, so the above link is the only evidence that it doesn't break the build.

### CODE REVIEW

**Comment by Rafer45** [comment]
> LGTM but please get another pair of eyes from infra on this PR, as my experience with build-system is limited and we wanna be sure we're not missing anything important

I'm not authorized to add other reviewers on this repo, could you do it?

File: build-system/tasks/runtime-test/runtime-test-base.js [hunk 1]
```diff
@@ -169,6 +169,14 @@ function updateReporters(config) {
   if (argv.saucelabs) {
     config.reporters.push('saucelabs');
   }
+
+  const testTypeAllowList = ['unit', 'integration'];
```
Comment (rcebulko) [inline build-system/tasks/runtime-test/runtime-test-base.js:173]: nit: wrap the array in `new Set()` and use `allowedTestTypes.has(config.testType)`. Approving to unblock

File: build-system/tasks/runtime-test/runtime-test-base.js [hunk 2]
```diff
@@ -169,6 +169,14 @@ function updateReporters(config) {
   if (argv.saucelabs) {
     config.reporters.push('saucelabs');
   }
+
+  const testTypeAllowList = new Set(['unit', 'integration']);
```
Comment (danielrozenberg) [inline build-system/tasks/runtime-test/runtime-test-base.js:173]: extreme bike-shedding: "allow list" doesn't seem right here (we allow all types of tests, we generate a JSON report for only some) - so how about calling it something like "JSON_REPORT_TEST_TYPES"?
(also, make this a module-level constant and make it all caps :) )

Comment (Rafer45) [reply build-system/tasks/runtime-test/runtime-test-base.js]: Done, I believe.

File: build-system/tasks/runtime-test/runtime-test-base.js (hunk 1 above)
Comment (Rafer45) [reply build-system/tasks/runtime-test/runtime-test-base.js]: Done.

**Pull request merged**